File: LeetCode/Easy/1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.go
func totalMoney(n int) int {
    ret := 0
    weeks := n / 7
    n %= 7

    for n > 0 {
        ret += n + weeks
        n--
    }

    weeks--;

    for weeks > -1 {
        ret += 28 + (weeks * 7)
        weeks--
    }

    return ret
}

File: LeetCode/Easy/1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.java
class Solution {
    public int totalMoney(int n) {
        int ret = 0;
        int weeks = n / 7;
        n %= 7;
        
        while (n > 0){
            ret += n + weeks;
            n--;
        }
        
        weeks--;
        
        while (weeks > -1){
            ret += 28 + (weeks * 7);
            weeks--;
        }
        
        return ret;
    }
}